doc/tutorials/core/mat-mask-operations/mat_mask_operations.markdown

```diff
@@ -104,26 +283,78 @@ Result.row(Result.rows - 1).setTo(Scalar(0)); // The bottom row
+in these points and, for example, set the pixels on the borders to zeros:
+@code{.java}
+        Result.row(0).setTo(new Scalar(0));
+        Result.row(Result.rows()-1).setTo(new Scalar(0));
+        Result.col(0).setTo(new Scalar(0));
+        Result.col(Result.cols()-1).setTo(new Scalar(0));
+@endcode
+@end_toggle
 
 The filter2D function
 ---------------------
 
 Applying such filters are so common in image processing that in OpenCV there exist a function that
 will take care of applying the mask (also called a kernel in some places). For this you first need
-to define a *Mat* object that holds the mask:
+to define an object that holds the mask:
+@add_toggle{C++}
 @code{.cpp}
 Mat kern = (Mat_<char>(3,3) <<  0, -1,  0,
                                -1,  5, -1,
                                 0, -1,  0);
 @endcode
-Then call the @ref cv::filter2D function specifying the input, the output image and the kernell to
+@end_toggle
+@add_toggle{Java}
+@code{.java}
+Mat kern = new Mat( 3, 3, CvType.CV_8U );
+int row = 0, col = 0;
+kern.put(row ,col, 0, -1, 0, -1, 5, -1, 0, -1, 0 );
+@endcode
+@end_toggle
+@add_toggle{Python}
+@code{.py}
+kernel = np.array([ [0,-1,0],
+                    [-1,5,-1],
+                    [0,-1,0] ],np.float32) # kernel should be floating point type
+@endcode
+@end_toggle
+
+@add_toggle{C++}
+Then call the @ref cv::filter2D function specifying the input, the output image and the kernel to
 use:
 @code{.cpp}
 filter2D(I, K, I.depth(), kern);
 @endcode
-The function even has a fifth optional argument to specify the center of the kernel, and a sixth one
-for determining what to do in the regions where the operation is undefined (borders). Using this
-function has the advantage that it's shorter, less verbose and because there are some optimization
+The function even has a fifth optional argument to specify the center of the kernel, a sixth
+for adding an optional value to the filtered pixels before storing them in K and a seventh one
+for determining what to do in the regions where the operation is undefined (borders).
+@end_toggle
+
+@add_toggle{Java}
+Then call the [Imgproc.filter2D()] function specifying the input, the output image and the kernel to
+use:
+@code{.java}
+Imgproc.filter2D(I, K, I.depth(), kern );
+@endcode
+The function even has a fifth optional argument to specify the center of the kernel, a sixth
+for adding an optional value to the filtered pixels before storing them in K and a seventh one
+for determining what to do in the regions where the operation is undefined (borders).
+@end_toggle
+
+@add_toggle{Python}
+Then call the @ref cv::filter2D function specifying the input, the output image and the kernell to
+use:
+@code{.py}
+K = cv2.filter2D(I, -1, kernel) # ddepth = -1, means destination image has depth same as input image.
+@endcode
+@end_toggle
+
+Using this function has the advantage that it's shorter, less verbose and because there are some optimization
```

Comment:
This function is shorter, less verbose and, because there are some optimizations, it is usually faster ...